### PR TITLE
Add inline class -X flag

### PR DESCRIPTION
**Pull request by jeffzoch**
https://github.com/JetBrains/kotlin/blob/68fdeaf865f279cb7a75eaec7e9f4e6cd0646707/compiler/testData/cli/jvm/extraHelp.out#L141

### CODE REVIEW

File: kotlin/internal/opts.bzl [hunk 1]
```diff
@@ -49,6 +49,27 @@ _KOPTS = {
...
+        type = attr.bool,
+    ),
+    "x_allow_result_return_type": struct(
+        args = dict(
+            default = False,
+            doc = "Enable result in return types",
+        ),
+        type = attr.bool,
+    ),
+    "x_opt_in": struct(
```
Comment (restingbull) [inline kotlin/internal/opts.bzl:66]: See https://github.com/bazelbuild/rules_kotlin/pull/398#pullrequestreview-545091307 -- opt ins appear to be a an area of growth, so the api ought to considered.

File: kotlin/internal/opts.bzl [hunk 2]
```diff
@@ -49,6 +49,27 @@ _KOPTS = {
...
+        args = dict(
+            default = False,
+            doc = "Enable experimental inline classes",
+        ),
+        type = attr.bool,
+    ),
+    "x_allow_result_return_type": struct(
+        args = dict(
+            default = False,
+            doc = "Enable result in return types",
```
Comment (restingbull) [inline kotlin/internal/opts.bzl:62]: Needs a bit more documentation -- took a minute to remember what Result was.

**Review by restingbull — CHANGES_REQUESTED** [review]
Please remove the gradle files.

File: kotlin/internal/opts.bzl (hunk 1 above)
Comment (jeffzoch) [reply kotlin/internal/opts.bzl:66]: Let me reopen this PR on a branch - just realized this is off my master

**Comment by jeffzoch** [comment]
closed in favor of #421

**Pull request closed without merging**